perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -351,43 +354,88 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
+      (all_streams_start + all_streams_duration))
+  records_in_interval = records[
+      analyze.FullyInInterval(records['start_time'],
+                              records['latency'],
+                              all_streams_start,
+                              all_streams_duration)]
+
   # Don't publish the full distribution in the metadata because doing
   # so might break regexp-based parsers that assume that all metadata
   # values are simple Python objects. However, do add an
   # 'object_size_B' metadata field even for the full results because
   # searching metadata is easier when all records with the same metric
   # name have the same set of metadata fields.
-  overall_metadata['object_size_B'] = 'distribution'
+  distribution_metadata = metadata.copy()
+  distribution_metadata['object_size_B'] = 'distribution'
+
+  latency_prefix = 'Multi-stream %s latency ' % operation
+  throughput_prefix = 'Multi-stream %s throughput ' % operation
   logging.info('Processing %s multi-stream %s results for the full '
                'distribution.', len(records), operation)
-  _AppendPercentilesToResults(
-      results,
-      (record['latency'] for record in records),
-      metric_name,
+  analyze.AppendStatsAsSamples(
+      records_in_interval['latency'],
       'sec',
-      overall_metadata)
+      results,
+      name_prefix=latency_prefix,
+      metadata=distribution_metadata)
 
+  analyze.AppendStatsAsSamples(
+      records_in_interval['size'] / records_in_interval['latency'],
+      'B/sec',
+      results,
+      name_prefix=throughput_prefix,
+      metadata=distribution_metadata)
+
+  logging.info('Processing %s multi-stream %s results for net throughput',
+               len(records), operation)
+  results.append(
+      sample.Sample(
+          'Multi-stream %s net throughput' % operation,
+          records_in_interval['size'].sum() /
+          records_in_interval['latency'].sum(),
+          'B/sec',
+          metadata=distribution_metadata))
+
+
+  # Publish by-size and full-distribution stats even if there's only
+  # one size in the distribution, because it simplifies postprocessing
+  # of results.
   for size in sizes:
-    metadata = metadata.copy()
-    metadata['object_size_B'] = size
+    this_size_records = records_in_interval[records_in_interval['size'] == size]
+    this_size_metadata = metadata.copy()
+    this_size_metadata['object_size_B'] = size
     logging.info('Processing %s multi-stream %s results for object size %s',
                  len(records), operation, size)
-    _AppendPercentilesToResults(
-        results,
-        (record['latency'] for record in records if record['size'] == size),
-        metric_name,
+    analyze.AppendStatsAsSamples(
+        this_size_records['latency'],
         'sec',
-        metadata)
+        results,
+        name_prefix=latency_prefix,
+        metadata=this_size_metadata)
+
+    analyze.AppendStatsAsSamples(
```

Comment:
For throughput percentiles, I think one reason they can be confusing is that per-object percentiles don't represent the actual throughput you will see, because a large and a small object count the same amount in the percentiles, but you will spend much more time uploading a large object than a small object. What do you think of weighting the per-object throughput by the number of seconds it took to upload that object? (I think this would be a separate PR.)

It would also be possible to use the code in `perfkitbenchmarker/traces/dstat.py` to run dstat on the worker VM and get an empirical view of the same data.

Either of these would eliminate the issue of object sizes, but they might make the throughput data less easy to understand. What do you think?